plugins/fclrave/fclspace.cpp

```diff
@@ -569,9 +588,14 @@ void FCLSpace::_ResetGeometryGroupsCallback(boost::weak_ptr<FCLKinBodyInfo> _pin
     if( !!pinfo ) {// && pinfo->_geometrygroup.size() > 0 ) {
         //RAVELOG_VERBOSE_FORMAT("env=%d, resetting geometry groups for kinbody %s, nGeometryUpdateStamp=%d (key %s, self=%d)", _penv->GetId()%pbody->GetName()%pinfo->nGeometryUpdateStamp%_userdatakey%_bIsSelfCollisionChecker);
         pinfo->nGeometryUpdateStamp++;
-        FCLKinBodyInfoRemover remover(boost::bind(&FCLSpace::RemoveUserData, this, pbody)); // protect
-        InitKinBody(pbody, pinfo, false);
-        remover.ResetRemove(); // succeeded
+        try {
+            ReloadKinBodyLinks(pbody, pinfo);
+        }
+        catch (...) {
+            // If something goes wrong processing this body, remove it from the FCL space.
+            RemoveUserData(pbody);
```

Comment:
@rschlaikjer Isn't the RAII remover much cleaner? Why change to this?

Reply:
If you don't want the bind cost, why not make a specialized remover class?

Reply:
(I found something like cancellable boost_exit but I could not find one)